packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -579,6 +580,7 @@ class _CupertinoBackGestureController<T> {
     @required this.navigator,
     @required this.controller,
     @required this.onEnded,
+    @required this.route,
```

Comment:
since this is private, we can just make it take route (and make it impossible for future maintainers to accidentally give in disjoint instances of these 3 things).

Reply:
Good idea, done.